### ISSUE TITLE
[BUG] `Azure.Monitor.Query` can't read `NaN`, `+∞` or `-∞`

### ISSUE THREAD

**Issue by afscrome** [Azure/azure-sdk-for-net#24975]
**Describe the bug**
If a query returns real values of `NaN`, `+∞` or `-∞`, then the query client is unable to read those rows -complaining that the result is a string, not a number

**Expected behavior**
I expected these values to be parsed as `double.NaN`, `double.PositiveInfinity` and `double.NegativeInfinity` respsectively

**Actual behaviour (include Exception or Stack Trace)**

```
System.InvalidOperationException: The requested operation requires an element of type 'Number', but the target element has type 'String'.
   at System.Text.Json.JsonDocument.TryGetValue(Int32 index, Double& value)
   at System.Text.Json.JsonElement.GetDouble()
   at Azure.Monitor.Query.Models.LogsTableRow.GetDouble(Int32 index)
   at Azure.Monitor.Query.Models.LogsTableRow.GetObject(Int32 index)
   at Azure.Monitor.Query.Models.LogsTableRow.get_Item(Int32 index)
```

**To Reproduce**

Run the below code

```cs
var client = new LogsQueryClient(_tokenCredential);

var results = await client.QueryWorkspaceAsync(workspaceId, "print real(123), real(nan), real(+inf), real(-inf), real(null)", default);
var table = results.Value.Table;

for (int i = 0; i < table.Columns.Count; i++)
{
    Console.Write($"Column {i}: ");
    try
    {
        Console.WriteLine(table.Rows[0][i]);
    }
    catch (Exception ex)
    {
        Console.WriteLine(ex);
    }
}
```

Expect to see the output similar to
```
Column 0: 123
Column 1: NaN
Column 2: ∞
Column 3: -∞
Column 4: null
```

But exceptions are logged for columns 1,2 and 3.

**Environment:**
 - `Azure.Monitor.Query` `1.0.0`
 -  netcoreapp3.1
 - IDE and version : Microsoft Visual Studio Professional 2019 Version 16.11.4

**Comment by nisha-bhatia** [comment]
This bug has been resolved and will go out in the next release. Please re-open if there are any further issues.

**Comment by scottaddie** [comment]
@afscrome This fix is available in the 1.0.1 release, which shipped yesterday.
